contrib/trace_propagators/src/test/java/io/opentelemetry/contrib/trace/propagation/AwsXRayPropagatorTest.java

```diff
@@ -0,0 +1,214 @@
+
+  private static final String TRACE_ID_FIRST_PART = "8a3c60f7";
+  private static final String TRACE_ID_SECOND_PART = "d188f8fa79d48a391a778fa6";
+  private static final String TRACE_ID_BASE16 = TRACE_ID_FIRST_PART + TRACE_ID_SECOND_PART;
+  private static final TraceId TRACE_ID = TraceId.fromLowerBase16(TRACE_ID_BASE16, 0);
+  private static final String TRACE_ID_XRAY =
+      "1-" + TRACE_ID_BASE16.substring(0, 8) + "-" + TRACE_ID_BASE16.substring(8);
+  private static final TraceState TRACE_STATE_DEFAULT = TraceState.builder().build();
+
+  private static final String SPAN_ID_BASE16 = "53995c3f42cd8ad8";
+  private static final SpanId SPAN_ID = SpanId.fromLowerBase16(SPAN_ID_BASE16, 0);
+  private static final TraceFlags SAMPLED_TRACE_FLAG =
+      TraceFlags.builder().setIsSampled(true).build();
+
+  private static final HttpTextFormat.Setter<Map<String, String>> setter =
+      new HttpTextFormat.Setter<Map<String, String>>() {
+        @Override
+        public void set(Map<String, String> carrier, String key, String value) {
+          carrier.put(key, value);
+        }
+      };
+  private static final HttpTextFormat.Getter<Map<String, String>> getter =
+      new HttpTextFormat.Getter<Map<String, String>>() {
+        @Nullable
+        @Override
+        public String get(Map<String, String> carrier, String key) {
+          return carrier.get(key);
+        }
+      };
+  private final AwsXRayPropagator xrayPropagator = new AwsXRayPropagator();
+
+  @Test
+  public void inject_SampledContext() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    xrayPropagator.inject(
+        withSpanContext(
+            SpanContext.create(TRACE_ID, SPAN_ID, SAMPLED_TRACE_FLAG, TRACE_STATE_DEFAULT),
+            Context.current()),
+        carrier,
+        setter);
+
+    assertThat(carrier)
+        .containsEntry(
+            TRACE_HEADER_KEY, "Root=" + TRACE_ID_XRAY + ";Parent=" + SPAN_ID_BASE16 + ";Sampled=1");
+  }
+
+  @Test
+  public void inject_NotSampledContext() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    xrayPropagator.inject(
+        withSpanContext(
+            SpanContext.create(TRACE_ID, SPAN_ID, TraceFlags.getDefault(), TRACE_STATE_DEFAULT),
+            Context.current()),
+        carrier,
+        setter);
+    assertThat(carrier)
+        .containsEntry(
+            TRACE_HEADER_KEY, "Root=" + TRACE_ID_XRAY + ";Parent=" + SPAN_ID_BASE16 + ";Sampled=0");
+  }
+
+  @Test
+  public void extract_SampledContext() {
+    Map<String, String> carrier = new LinkedHashMap<>();
+    carrier.put(
+        TRACE_HEADER_KEY, "Root=" + TRACE_ID_XRAY + ";Parent=" + SPAN_ID_BASE16 + ";Sampled=1");
+
+    assertThat(getSpanContext(xrayPropagator.extract(Context.current(), carrier, getter)))
+        .isEqualTo(
+            SpanContext.createFromRemoteParent(
+                TRACE_ID, SPAN_ID, SAMPLED_TRACE_FLAG, TRACE_STATE_DEFAULT));
+  }
+
+  @Test
+  public void extract_NotSampledContext() {
```

Comment:
Couple more test cases

1) Valid header with keys in different oder
2) Valid header with additional params. We strip this right now but need to make sure this propagator can read any valid header

Reply:
Ah can we also add a test where we inject and there's some trace state? Documents that we ignore for now and will give a target to update when actually implementing it.

Reply:
Sure, will add those tests